discovery/scaleway/baremetal.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/common/model"
+	"github.com/scaleway/scaleway-sdk-go/api/baremetal/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	baremetalLabelPrefix = metaLabelPrefix + "baremetal_"
+
+	baremetalIDLabel        = baremetalLabelPrefix + "id"
+	baremetalZoneLabel      = baremetalLabelPrefix + "zone"
+	baremetalNameLabel      = baremetalLabelPrefix + "name"
+	baremetalStatusLabel    = baremetalLabelPrefix + "status"
+	baremetalTagsLabel      = baremetalLabelPrefix + "tags"
+	baremetalProjectIDLabel = baremetalLabelPrefix + "project_id"
```

Comment:
maybe add the ip?

Reply:
We can have a variable number of IPs in a server. I'm not sure about how to handle that with label @roidelapluie

Reply:
How do you identify the IP addresses? Do they belong to networks we can identify by name?

Reply:
are they assigned with networks which could have names or uuid's?

Reply:
Not at the moment, it could come later on with VPC but it is not currently the case. Flexible IP can be added: https://www.scaleway.com/en/docs/configuring-multiple-flexible-ips/

Reply:
I've removed the IP labels altogether. I got two labels: IP version and is_ip_public. it is the same across baremetal and instance